### ISSUE TITLE
Spike] investigate moving dynamic profile to statically generated profile in order to avoid emf compare problem

### ISSUE THREAD

**Issue by ysroh** [ZeligsoftDev/CX4CBDDS#221]
# **Issue and tracking information**

I am curious to know if all of the compare/merge problems will be gone if we were to migrate our profile to static profile.
Static or dynamic profile will results in the same UML content which means the content of UML file does not change. The difference is how stereotype applications are loaded and accessed during the runtime. Theoretically, you should be able to switch dynamic profile to static and everything should still work assuming that you are just using vanilla Eclipse tooling (Properties view, menus etc..). However, our tooling will require some makeup in order to support static profile.

According to Ernesto the memory leak problem didn't even happen in static profile situation so I am guessing that the other problem might be due to dynamic profile as well. We can at least see if that's the case or not.

If you approve this work then we can try to switch our profile to static and see the differences. Theoretically, Papyrus should load the existing model without any problem so it won't be a problem to test compare merge scenario. 

I think it would be worthwhile to investigate this since it might give us an alternative solution to the merge problem we are trying to solve and we don't need to maintain different version of EMF compare if it works.

### Developer's Actual time spent on fix (hours)  

# **Issue reporter to provide a detailed description of the issue in the space below**

**Comment by ysroh** [comment]
I've tried to switch our profile to static profile and tested Papyrus compare merge.
As far as I can tell I experienced no crash nor any NPEs during the test. Everything seems to be smooth. I doubt that the result would be any different on your environment but I strongly suggest this to be tested for your scenario.
The changes are pushed to papyrus_static_no_emf_compare branch. You do not need to have the new EMF compare you can safely uninstall it if already installed.

The changes do not contain updates for tooling fix so some of the tooling and generation do not work yet including CX properties view. You cannot create a new Model either. Modification of diagrams and UML and Profile properties view are working so you can still make changes to the model. There shouldn't be any problem to use this to test compare/merge scenario.

You need to first open the existing model and click OK when profile migration dialog is open.
In order to use compare/merge you need to check in this version to git since you cannot compare two different models with different profile type.

I would estimate around 2-3 weeks to complete the transition from dynamic to static profile.

Please let me know if you have any questions.

![image](https://user-images.githubusercontent.com/896503/111002953-44d17000-8354-11eb-8e98-24b63ca5ebb7.png)

**Comment by j26151** [comment]
When attempting to migrate one of our models, I get the following: 

![image](https://user-images.githubusercontent.com/49879646/117142573-dae6ad00-ad7d-11eb-8e85-efc05a72edb6.png)

**Comment by ysroh** [comment]
@j26151  Can you let me know if this IDLPubSub model is using any other libraries and they are all migrated as well??

**Comment by j26151** [comment]
This is happening for more than just the IDLPubSub model. The IDLPubSub model only includes the DDS4CCM library as far as i know and neither of them are able to migrate. In the ".uml" file I can see the following:

![image](https://user-images.githubusercontent.com/49879646/117144393-dd4a0680-ad7f-11eb-8ee5-f91665b7b918.png)

**Comment by ysroh** [comment]
Is it possible to send me the model so I can try here??
Also can you please try to see if you can migrate BasicPubSub model? Maybe also try migrate from RSA to Papyrus not that is required but want to know if what is wrong.

Can you also let me know what other models exist in your workbench?? Can you close all other projects and leave this project only then restart your workbench and try again??


Thanks.

**Comment by j26151** [comment]
Sorry about all of the confusion. It appears that something was messed up in my PapyrusXcx-axcioma installation. I wiped my system clean and reinstalled everything, and it appears to migrate properly now. Thank you for the help. I am going to verify a few things on our end and then let you know if this is good to close from our side.

**Comment by ysroh** [comment]
Great!!! Thanks

**Comment by j26151** [comment]
This all looks good to me. I was able to generate IDL and descriptors that look correct as well as perform a model merging scenario. I did still receive 2 separate NPEs. I don't know how to recreate them, but it would lead me to believe that there are still smaller issues between the git perspective, merging, and other modeling things within papyrus. Overall though, every thing worked quickly and the NPEs were able to be ignored.
